### ISSUE TITLE
Number of relaychain peers isn't limited 

### ISSUE THREAD

**Issue by stakeworks** [darwinia-network/darwinia#1182]
The max number of relaychain peers should be 40 with default settings:

      --out-peers <COUNT>
          Specify the number of outgoing connections we're trying to maintain
          [default: 8]

      --in-peers <COUNT>
          Maximum number of inbound full nodes peers
          [default: 32]

But this doesn't work, also explicit setting in-peers and out-peers for relaychain doesn't work.

Crab
```
2023-06-16 10:59:16 [Relaychain] 💤 Idle (413 peers), best: #18379961 (0x6165…ee5b), finalized #18379958 (0x1596…09c7), ⬇ 462.1kiB/s ⬆ 555.8kiB/s

```

Darwinia
```
2023-06-16 11:24:47 [Relaychain] 💤 Idle (876 peers), best: #15990958 (0x5a22…075f), finalized #15990956 (0xd094…7afe), ⬇ 118.4kiB/s ⬆ 306.1kiB/s

```

Ubuntu 20.04.6 LTS & Ubuntu 22.04 LTS
darwinia 6.3.1-unknown =>

**Comment by AurevoirXavier** [comment]
What's your boot command?

**Comment by stakeworks** [comment]
Same thing happened with the Interlay Kintsugi chain, after the update to Polkadot 0.9.38

**Comment by AurevoirXavier** [comment]
> Same thing happened with the Interlay Kintsugi chain, after the update to Polkadot 0.9.38

I think there are a lot of light nodes.

Try `--in-peers-light 35`.

**Comment by stakeworks** [comment]
I've tried this:

        --out-peers 15 \
        --in-peers 35 \
        --in-peers-light 35 \

But after a few minutes I already have over 300 peers.

2023-06-24 09:32:46 [Relaychain] 💤 Idle (309 peers),

**Comment by AurevoirXavier** [comment]
Tracking https://github.com/paritytech/cumulus/issues/2563

**Comment by altonen** [comment]
This issue isn't related to light clients. There was a problem in 0.9.40 where the network event stream got clogged and displayed the number of peers incorrectly. This was fixed in 0.9.41
